### ISSUE TITLE
IDBObjectStore»add() example: incomplete basic explanation; unclear how ConstraintError is handled

### ISSUE THREAD

**Issue by SoftMoonWebWare** [mdn/content#15037]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/add

### What specific section or headline is this issue about?

Errors, Exceptions

### What information was incorrect, unhelpful, or incomplete?

It is unclear how  ConstraintError  is handled.
Text says "If a record already exists in the object store with the key parameter as its key, then an error ConstrainError event is fired on the returned request object."
OK.  How do I tell in my "onerror" method if it is a ConstraintError or otherwise?  Is the "event" object passed to the "onerror" method an instance of Error?  Event?  hybrid?  Error a property of Event?
Text says "This method may raise a [DOMException] of one of the following types:"
OK.  When?  The example does not handle a DOMException, just an "onerror" method.
In fact, the example confusingly applies the "onerror" to the transaction, not the request object as stated (yea, I know it bubbles to the transaction, but the example is still confusing if that is not noted to the beginner)
If I could: 
try { transaction.objectStore(store).add(foo); }
catch(e) { (e instanceof ConstraintError) ? ......... }
then I could move forward.
It's a lot of work to create a database, fill it, then try to create errors and probe results from there.

### What did you expect to see?

a complete reference to the method

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/idbobjectstore/add`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/add
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/idbobjectstore/add/index.md
* Last commit: https://github.com/mdn/content/commit/281e3b21178946c8301232a8eb50d11770ee8450
* Document last modified: 2022-02-18T09:53:31.000Z

</details>

**Comment by SoftMoonWebWare** [comment]
A simple cross-reference link to DOMException would have sufficed.
It seems the reference pages are being dumbed down.  While I can't demonstrate a specific example, I used to have no problem using this site to not only reference JavaScript, but to learn it.  Now the pages are so sparse, cross-reference links are fewer, technical details are not fully explained.  Things that I could quickly find before, I have to search and search and search for.  Like I looked for quite a while to find the docs on DOMException.  I did finally find them, but only after a long search and then posting this note.
My point is this site is loosing its helpfulness.  No, don't explain how to USE DOMExceptions on this page, but a simple example would be nice.  A **link** to DOMException docs would be even more relavant.  No extra text in the page, just a link on the existing text.
Yesterday I needed the exact specs for using @include in a CSS file.  Had to SEARCH for it for five mins.  And I know the basic structure of the site, because I'm a nerd and go through it for fun on Friday nights looking for the latest new info.  If I was a nooby, wow, I'd be lost.
